spec/services/checksum_validator_spec.rb

```diff
@@ -208,27 +208,213 @@
   end
 
   context '#validate_checksums' do
-    context 'passes validation' do
+    context 'passes checksum validation' do
       let(:druid) { 'bz514sm9647' }
       let(:endpoint_name) { 'fixture_sr1' }
 
       it 'returns a positive result for a pres_copy' do
         cv.validate_checksums
-        expect(cv.checksum_results.result_array.first).to have_key(:moab_checksum_valid)
+        expect(cv.results.result_array.first).to have_key(:moab_checksum_valid)
+      end
+
+      [
+        PreservedCopy::ONLINE_MOAB_NOT_FOUND_STATUS,
+        PreservedCopy::INVALID_MOAB_STATUS,
+        PreservedCopy::UNEXPECTED_VERSION_ON_STORAGE_STATUS,
+        PreservedCopy::INVALID_CHECKSUM_STATUS,
+        PreservedCopy::VALIDITY_UNKNOWN_STATUS
+      ].each do |initial_status|
+        it "sets status to OK_STATUS if it was previously #{initial_status}" do
+          pres_copy.status = initial_status
+          pres_copy.save!
+          expect { cv.validate_checksums }.to change { pres_copy.status }.to PreservedCopy::OK_STATUS
+          expect(pres_copy.reload.status).to eq PreservedCopy::OK_STATUS
+        end
+      end
+
+      it "leaves status as OK_STATUS as-is" do
```

Comment:
Nitpicky, but this reads to me "leaves status of OK_STATUS as-is", but not sure if that was your intent.

Reply:
good catch, definitely a typo, artifact of repeated editing.

Reply:
fixed and repushed